lnrpc/routerrpc/router_backend.go

```diff
@@ -547,6 +547,14 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 	}
 	payIntent.CltvLimit = cltvLimit
 
+	// Take max htlcs from the request. Map zero to one for backwards
+	// compatibility.
+	maxHtlcs := rpcPayReq.MaxHtlcs
```

Comment:
should be done in an earlier commit?

Reply:
squashed most of this and only left the `lncli` changes as a separate commit.